syntax = "proto3";

package Protobuf;
option java_multiple_files = true;

message Ball {
    double x = 1;
    double y = 2;
    double z = 3;
    double vx = 4;
    double vy = 5;
    double vz = 6;
}

message Robot {
    uint32 robot_id = 1;
    double x = 2;
    double y = 3;
    double orientation = 4;
    double vx = 5;
    double vy = 6;
    double vorientation = 7;
}

message Field {
    double width = 1;
    double length = 2;
    double goal_width = 3;
    double goal_depth = 4;
}

message Frame {
    Ball ball = 1;
    repeated Robot robots_yellow = 2;
    repeated Robot robots_blue = 3;
}
message Environment {
	uint32 step = 1;
	Frame frame = 2;
	Field field = 3;
	uint32 goals_blue = 4;
	uint32 goals_yellow =5;
}

message RobotReplacement {
	Robot  position   = 1;
	bool   yellowteam = 5;
	bool   turnon     = 6;
}

message BallReplacement {
	double x  = 1;
	double y  = 2;
	double vx = 3;
	double vy = 4;
}

message Replacement {
	BallReplacement           ball   = 1;
	repeated RobotReplacement robots = 2;
}

message Command {
	uint32 id          = 1;
	bool   yellowteam  = 2;
	double wheel_left  = 6;
	double wheel_right = 7;
}

message Commands {
	repeated Command robot_commands = 1;
}

message Packet {
	Commands    cmd     = 1;
	Replacement replace = 2;
}
